lib/Myriad/Exception/BadMessage.pm

```diff
@@ -0,0 +1,17 @@
+package Myriad::Exception::BadMessage;
+
+use strict;
+use warnings;
+
+# VERSION
+
+use parent qw(Myriad::Exception);
+
+no indirect;
+
+sub throw {
+    my ($self, $field) = @_;
```

Comment:
It would probably be a lot more self-documenting at callsites to pass this by a named param

```sub throw ($self, %args) {
    $self->SUPER::throw("Bad RPC Message field: $args{field} is required", "BAD_MESSAGE");
}
```

then elsewhere

```perl
   Myriad::Exception::BadMessage->throw(field => "addr");
```